st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -469,6 +469,58 @@ def post(self, liveaction_api, requester_user, context_string=None, show_secrets
                                                context_string=context_string,
                                                show_secrets=show_secrets)
 
+    def put(self, id, liveaction_api, requester_user, show_secrets=False):
+        """
+        Updates a single execution.
+
+        Handles requests:
+            PUT /executions/<id>
+
+        """
+        if not requester_user:
+            requester_user = UserDB(cfg.CONF.system_user.user)
+
+        from_model_kwargs = {
+            'mask_secrets': self._get_mask_secrets(requester_user, show_secrets=show_secrets)
+        }
+
+        execution_api = self._get_one_by_id(id=id, requester_user=requester_user,
+                                            from_model_kwargs=from_model_kwargs,
+                                            permission_type=PermissionType.EXECUTION_STOP)
+
+        if not execution_api:
+            abort(http_client.NOT_FOUND, 'Execution with id %s not found.' % id)
+
+        liveaction_id = execution_api.liveaction['id']
+        if not liveaction_id:
+            abort(http_client.INTERNAL_SERVER_ERROR,
+                  'Execution object missing link to liveaction %s.' % liveaction_id)
+
+        try:
+            liveaction_db = LiveAction.get_by_id(liveaction_id)
+        except:
+            abort(http_client.INTERNAL_SERVER_ERROR,
```

Comment:
The delete method is already using 500 for similar cases at https://github.com/StackStorm/st2/blob/master/st2api/st2api/controllers/v1/actionexecutions.py#L501. Also, it's technically not 404 because the execution which the user is updating/deleting is found but the reference to liveaction in the execution object is missing. That's an internal system error for whatever reason.